### PR TITLE
Added support for querying the architecture of the toit vm.

### PR DESCRIPTION
**Pull request by mikkeldamsgaard**

### CODE REVIEW

File: lib/core/utils.toit [hunk 1]
```diff
@@ -256,6 +256,19 @@ PLATFORM-WINDOWS ::= "Windows"
 PLATFORM-MACOS ::= "macOS"
 PLATFORM-LINUX ::= "Linux"
 
+/** Returns a string identifying the underlying architecture. */
+architecture:
```
Comment (erikcorry) [inline lib/core/utils.toit:260]:
```suggestion
architecture -> string:
```

File: src/os_posix.cc [hunk 2]
```diff
@@ -308,6 +308,20 @@ size_t ProtectableAlignedMemory::compute_alignment(size_t alignment) {
...
+#if defined(__aarch64__)
+  return "arm64";
+#elif defined(__arm__)
+  return "arm";
+#elif defined(__amd64__)
+  return "x86_64";
+#elif defined(__i386__)
+  return "x86";
+#else
+  return "unkonwn";
```
Comment (erikcorry) [inline src/os_posix.cc:321]:
```suggestion
  #error "Unknown CPU architecture"
```

File: src/os_win.cc [hunk 3]
```diff
@@ -366,6 +366,18 @@ const char* OS::get_platform() {
...
 
+const char* get_architecture() {
+#if defined(_M_AMD64)
+  return "x86_64"
+#elif defined(_M_ARM64)
+  return "arm64";
+#elif defined(_M_IX86)
+  return "x86";
+#else
+  return "unknown";
```
Comment (erikcorry) [inline src/os_win.cc:377]:
```suggestion
  #error "Unknown CPU architecture"
```

**Review by erikcorry — APPROVED** [review]

File: src/os_esp32.cc [hunk 4]
```diff
@@ -496,6 +496,20 @@ const char* OS::get_platform() {
...
+#if defined(CONFIG_IDF_TARGET_ESP32S3)
+  return "esp32s3";
+#elif defined(CONFIG_IDF_TARGET_ESP32C3)
+  return "esp32c3";
+#elif defined(CONFIG_IDF_TARGET_ESP32S2)
+  return "esp32s2";
+#elif defined(CONFIG_IDF_TARGET_ESP32)
+  return "esp32";
+#else
+  #error "Unknown MCU architecture"
```
Comment (kasperl) [inline src/os_esp32.cc:509]: Not sure if it is really helpful that the unknown case yields two different strings (CPU <> MCU). Maybe just "Unknown architecture"?

**Comment by erikcorry** [comment]
I resolved the conflicts in origin/arch-for-hosts.  I think if you merge up to that, then push to your own repo then the PR will update accordingly.

**Pull request merged**